### PR TITLE
bpo-43612: Add wbits parameter to zlib.compress

### PR DESCRIPTION
**Pull request by rhpvorderman**
This allows users to compress raw deflate blocks without zlib headers and trailers in one go. Without needing to use zlib.compressobj. Also this allows users to do `zlib.compress(data, 1, wbits=31)` which is much faster than `gzip.compress(data, 1)`. 

https://bugs.python.org/issue43612

### CODE REVIEW

File: Doc/library/zlib.rst [hunk 1]
```diff
@@ -47,19 +47,43 @@ The available exception and functions in this module are:
       platforms, use ``adler32(data) & 0xffffffff``.
 
 
-.. function:: compress(data, /, level=-1)
+.. function:: compress(data, /, level=-1, wbits=MAX_WBITS)
 
    Compresses the bytes in *data*, returning a bytes object containing compressed data.
+
```
Comment (ZackerySpytz) [inline Doc/library/zlib.rst]: Unrelated whitespace changes usually aren't accepted.

File: Doc/library/zlib.rst [hunk 2]
```diff
@@ -47,19 +47,43 @@ The available exception and functions in this module are:
...
+   * +25 to +31 = 16 + (9 to 15): Uses the low 4 bits of the value as the
+     window size logarithm, while including a basic :program:`gzip` header
+     and trailing checksum in the output.
+
    Raises the :exc:`error` exception if any error occurs.
 
    .. versionchanged:: 3.6
       *level* can now be used as a keyword parameter.
 
+   .. versionchanged:: 3.10
```
Comment (ZackerySpytz) [inline Doc/library/zlib.rst]: This will have to be bumped to 3.11.

File: Doc/library/zlib.rst [hunk 3]
```diff
@@ -47,19 +47,43 @@ The available exception and functions in this module are:
...
+      *wbits* parameter added.
```
Comment (ZackerySpytz) [inline Doc/library/zlib.rst]: Please use a complete sentence.

File: Doc/library/zlib.rst [hunk 4]
```diff
@@ -76,23 +100,9 @@ The available exception and functions in this module are:
...
-
-   * −9 to −15: Uses the absolute value of *wbits* as the
-     window size logarithm, while producing a raw output stream with no
-     header or trailing checksum.
-
-   * +25 to +31 = 16 + (9 to 15): Uses the low 4 bits of the value as the
-     window size logarithm, while including a basic :program:`gzip` header
-     and trailing checksum in the output.
+   The *wbits* parameter controls the size of the history buffer (or the
+   "window size"), and what header and trailer format will be used  It has
```
Comment (ZackerySpytz) [inline Doc/library/zlib.rst]: It looks like there's a missing period here.

File: Modules/zlibmodule.c [hunk 5]
```diff
@@ -227,7 +229,7 @@ zlib_compress_impl(PyObject *module, Py_buffer *data, int level)
     zst.zalloc = PyZlib_Malloc;
     zst.zfree = PyZlib_Free;
     zst.next_in = ibuf;
-    int err = deflateInit(&zst, level);
+    int err =  deflateInit2(&zst, level, DEFLATED, wbits, DEF_MEM_LEVEL, Z_DEFAULT_STRATEGY);
```
Comment (ZackerySpytz) [inline Modules/zlibmodule.c]: Please limit lines to 79 characters (and remove the extra space before `deflateInit2`).

File: Misc/NEWS.d/next/Library/2021-03-24-09-40-02.bpo-43612.vMGZ4y.rst [hunk 6]
```diff
@@ -0,0 +1,5 @@
+``zlib.compress`` now accepts a wbits parameter which allows users to
```
Comment (ZackerySpytz) [inline Misc/NEWS.d/next/Library/2021-03-24-09-40-02.bpo-43612.vMGZ4y.rst]:
```rst
:func:`zlib.compress`
```

File: Doc/library/zlib.rst (hunk 1 above)
Comment (rhpvorderman) [reply Doc/library/zlib.rst]: Removed.

**Comment by rhpvorderman** [comment]
@ZackerySpytz Thank you for the review! I addressed your comments.

File: Misc/NEWS.d/next/Library/2021-03-24-09-40-02.bpo-43612.vMGZ4y.rst (hunk 6 above)
Comment (rhpvorderman) [reply Misc/NEWS.d/next/Library/2021-03-24-09-40-02.bpo-43612.vMGZ4y.rst]: ~I did this, but it makes the docs build fail.
Should the `:func:` be present in blurbs as well?~

Edit: Nevermind. I spotted the issue with the double backticks.

**Comment by rhpvorderman** [comment]
Changes incorporated in https://github.com/python/cpython/pull/27941
